### PR TITLE
[Dependabot] Add Dependabot

### PR DESCRIPTION
**Pull request by amogkam**
Adds Dependabot to repo to monitor and automatically update Tune dependencies.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    directory: "/python/requirements"
+    schedule:
+      # TODO(amogkam) change this to weekly after some initial validation.
+      interval: "daily"
+      # 8 PM
+      time: "20:00"
+      # Use Pacific Standard Time
+      timezone: "America/Los_Angeles"
+    commit-message:
+      prefix: "[Tune]"
```
Comment (richardliaw) [inline .github/dependabot.yml:16]: nit - [tune]

**Review by richardliaw — APPROVED** [review]
very nice -- let me know when I should merge this.

File: .github/dependabot.yml (hunk 1 above)
Comment (amogkam) [inline .github/dependabot.yml:16]:
```suggestion
      prefix: "[tune]"
```

Comment (amogkam) [reply .github/dependabot.yml]: Done!

**Comment by amogkam** [comment]
@richardliaw this is ready to merge now and then we should start seeing PRs tonight

**Pull request merged**